packages/flutter_tools/lib/src/commands/attach.dart

```diff
@@ -316,8 +317,17 @@ known, it can be explicitly provided to attach via the command-line, e.g.
       final Status discoveryStatus = _logger.startSpinner(
         timeout: const Duration(seconds: 30),
         slowWarningCallback: () {
-          // On iOS we rely on mDNS to find Dart VM Service. Remind the user to allow local network permissions on the device.
-          if (_isIOSDevice(device)) {
+          // On iOS we rely on mDNS to find Dart VM Service.
+          if (device is IOSSimulator) {
+            // mDNS on simulators stopped working in macOS 15.4.
+            // See https://github.com/flutter/flutter/issues/166333.
+            return 'The Dart VM Service was not discovered after 30 seconds. '
```

Comment:
do we also wanna ask user to click "allow" like the iOS device case? or is the permission implicitly allowed for simulator case?

Reply:
Permission is implicitly **_disallowed_** for simulators. It will always fail on macOS 15 (from what I can tell). There is no settings you can change or button you can click to make it work.

Reply:
> For macOS 15 only?
It appears so

Reply:
so for users on macOS 14-, will they be prompt? do we wanna provide them with instructions?

Reply:
Mmm so I'm not 100% confident that it's solely sue to macOS 15.4. There are some conflicting reports in the issue: https://github.com/flutter/flutter/issues/166333. I updated the instructions here to give both